### DISCUSSION TITLE
Struggling with 4.2.0 upgrade

### DISCUSSION THREAD

**Discussion by stuartoram** [mastodon/mastodon#27035] [General]
Hi,

I'm currently running v4.1.9 but struggling with the upgrade to 4.2.0 due to a whole host of issues with the dependency upgrades.
I managed to work through several issues but got stumped by

ActionView::Template::Error: Alias parsing was not enabled. To enable it, pass `aliases: true` to `Psych::load` or `Psych::safe_load`.

For now, I've had to revert the VM back to a snapshot prior to upgrade attempts.

Is there a guide anywhere to steps required to upgrade successfully? I've managed several previous updates with a few dependency updates with no issue, but this one is beyond me.

**Comment by stuartoram** [comment]
Right, I seem to be upgraded & working.
To be honest, I can't say exactly what got it working as it was a bit of trial & error until I made it through without errors.

Edit:
This is the sequence (with errors/output snipped) that finally seemed to work for me.

Welcome to Ubuntu 23.04 (GNU/Linux 6.2.0-33-generic x86_64)
20:26:08[mastodon@mastodon]~$su mastodon_svc
Password:
mastodon_svc@mastodon:/home/mastodon$ cd /home/mastodon_svc/live
mastodon_svc@mastodon:~/live$ RUBY_CONFIGURE_OPTS=--with-jemalloc rbenv install 3.2.2
mastodon_svc@mastodon:~/live$ bundle install
mastodon_svc@mastodon:~/live$ yarn install --frozen-lockfile
mastodon_svc@mastodon:~/live$ git fetch && git checkout v4.2.0
mastodon_svc@mastodon:~/live$ RAILS_ENV=production bundle exec rails assets:precompile
mastodon_svc@mastodon:~/live$ bundle install
mastodon_svc@mastodon:~/live$ RAILS_ENV=production bundle exec rails assets:precompile
mastodon_svc@mastodon:~/live$ yarn add -D compression-webpack-plugin
mastodon_svc@mastodon:~/live$ RAILS_ENV=production bundle exec rails assets:precompile
mastodon_svc@mastodon:~/live$ SKIP_POST_DEPLOYMENT_MIGRATIONS=true RAILS_ENV=production bundle exec rails db:migrate
mastodon_svc@mastodon:~/live$ su mastodon
Password:
20:50:35[mastodon@mastodon]live$sudo -u postgres psql
postgres=# ALTER DATABASE mastodon_production REFRESH COLLATION VERSION;
postgres-# \q
20:52:54[mastodon@mastodon]live$su mastodon_svc
Password:
mastodon_svc@mastodon:~/live$ SKIP_POST_DEPLOYMENT_MIGRATIONS=true RAILS_ENV=production bundle exec rails db:migrate
mastodon_svc@mastodon:~/live$ su mastodon
Password:
20:54:09[mastodon@mastodon]live$sudo systemctl restart mastodon-*
20:54:24[mastodon@mastodon]live$su mastodon_svc
Password:
mastodon_svc@mastodon:~/live$ RAILS_ENV=production bundle exec rails db:migrate
mastodon_svc@mastodon:~/live$ RAILS_ENV=production bin/tootctl search deploy --reset-chewy

**Comment by renchap** [comment]
The upgrade instructions are in [the release notes](https://github.com/mastodon/mastodon/releases/tag/v4.2.0).

I think your error came from the Ruby version you had installed. Mastodon 4.2 requires Ruby 3.0+ (Ruby 3.2.2 is recommended).

**Reply by stuartoram** [reply]
I had upgraded Ruby previously but I don't think it quite went to plan.
